File: DS-Algo-with-Java/Lecture-6/CountInputString.java
import java.util.Scanner;

public class CountInputString {
    public static void main(String[] args) {
        try (Scanner sc = new Scanner(System.in)) {
            System.out.print("Enter a string: ");
            String s = sc.nextLine();
            
            char[] ch = s.toCharArray();
            int c_count = 0;
            int s_count = 0;
            int n_count = 0;
            int o_count = 0;

            for (int i = 0; i < s.length(); i++) {
                if (Character.isLetter(ch[i])) {
                    c_count++;
                } else if (Character.isSpaceChar(ch[i])) {
                    s_count++;
                } else if (Character.isDigit(ch[i])) {
                    n_count++;
                } else {
                    o_count++;
                }
            }
            System.out.println("The string is: " + s);
            System.out.println("letter: " + c_count);
            System.out.println("space: " + s_count);
            System.out.println("number: " + n_count);
            System.out.println("other: " + o_count);

        }
    }
}

File: DS-Algo-with-Python/Lecture-6/CountInputString.py
s = input("Enter a string: ")
digit = 0
space = 0
alpha = 0
other = 0

for char in s:
    if char.isdigit():
        digit += 1
    elif char.isspace():
        space += 1
    elif char.isalpha():
        alpha += 1
    else:
        other += 1

print(f"The string is: {s}")
print(f"letter: {alpha}")
print(f"space: {space}")
print(f"number: {digit}")
print(f"other: {other}")
